src/RulesFactory.js

```diff
@@ -0,0 +1,69 @@
+import warning from 'warning'
+
+import RegularRule from './rules/RegularRule'
+import SimpleRule from './rules/SimpleRule'
+import KeyframeRule from './rules/KeyframeRule'
+import ConditionalRule from './rules/ConditionalRule'
+import FontFaceRule from './rules/FontFaceRule'
+
+export default class RulesFactory {
+  constructor() {
+    this.classes = {
+      '@charset': SimpleRule,
+      '@import': SimpleRule,
+      '@namespace': SimpleRule,
+      '@keyframes': KeyframeRule,
+      '@media': ConditionalRule,
+      '@supports': ConditionalRule,
+      '@font-face': FontFaceRule
+    }
+    this.regExp = this.buildRegExp()
+  }
+
+  /**
+   * Build regexp for matching rules.
+   *
+   * @return {RegExp}
+   * @api private
+   */
+  buildRegExp() {
```

Comment:
Instead of having a private method like this on the prototype, why not just have it as a function outside of the class?

```js
function buildClassRegExp(classes: Object): RegExp  {
  return new RegExp(`^${Object.keys(classes).join('|')}`)
}
```